lib/matplotlib/animation.py

```diff
@@ -826,10 +824,10 @@ def setup(self, fig, outfile, dpi, frame_dir=None):
 
         if not self.embed_frames:
             if frame_dir is None:
-                frame_dir = root + '_frames'
-            if not os.path.exists(frame_dir):
-                os.makedirs(frame_dir)
-            frame_prefix = os.path.join(frame_dir, 'frame')
+                frame_dir = Path(str(outfile.with_suffix('')) + '_frames')
+            if not frame_dir.exists():
+                frame_dir.mkdir(parents=True)
```

Comment:
```suggestion
            frame_dir.mkdir(parents=True, exist_ok=True)
```

Reply:
Except replacing both lines, not just the second one

Reply:
changed